### DISCUSSION TITLE
Building and Running the UFS Weather Model on HERA

### DISCUSSION THREAD

**Discussion by sanatcumar** [ufs-community/ufs-weather-model#1809] [Q&A] [answered]
Hi I am a new user and trying to build and run the UFS weather model on HERA.  I tried to follow the documentation in https://ufs-weather-model.readthedocs.io/en/ufs-v1.1.0/BuildingAndRunning.html but hitting a wall as the instructions seem outdated. 
 
Can any one point me to updated instructions or provide me with the steps for building and running the UFS weather model on HERA please?
Cheers and thanks
Sanat

**Comment by gspetro-NOAA** [comment, marked as answer]
Hi Sanat,

The most recently updated instructions are here: https://ufs-weather-model.readthedocs.io/en/latest/BuildingAndRunning.html. I can't guarantee that they're 100% up-to-date, since we are frequently merging new code into the HEAD of the `develop` branch, but these instructions should work much better than the instructions you were using. If you run into issues, please continue to post here, and we will help you to get the UFS WM up and running!

Best,
Gillian Petro | Advanced User Support Team

**Reply by sanatcumar** [reply]
Hi Gillian,
 Thank you very much for the response. I was able to compile ufs-v2.0.0 with "export CCPP_SUITES="FV3_GFS_v15p2" but with export CCPP_SUITES=FV3_GFS_v16beta, it fails with the following error
in ccpp_prebuild.err 
INFO: Logging level set to INFO
INFO: Found TYPEDEFS_NEW_METADATA dictionary in config, assume at least some data is in new metadata format
INFO: Parsing suite definition files ...
INFO: Parsing suite definition file FV3/ccpp/suites/suite_FV3_GFS_v16beta.xml ...
CRITICAL: Suite definition file FV3/ccpp/suites/suite_FV3_GFS_v16beta.xml not found.
ERROR: Parsing suite definition file suite_FV3_GFS_v16beta.xml failed.
Traceback (most recent call last):
  File "FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 814, in <module>
    main()
  File "FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 738, in main
    raise Exception('Parsing suite definition files failed.')
Exception: Parsing suite definition files failed.

In addition, If I switch the branch to "develop" ./build.sh fails with ./build.sh: line 18: CMAKE_FLAGS: unbound variable

 
Cheers and thanks
Sanat

**Reply by gspetro-NOAA** [reply]
Hi Sanat,

It looks like FV3_GFS_v16beta is not supported in the v2.0.0 code (see v2.0.0 suites [here](https://github.com/NOAA-EMC/fv3atm/tree/release/public-v2/ccpp/suites)). That said, there are a whole host of suites available in the `develop` branch (see [here](https://github.com/NOAA-EMC/fv3atm/tree/dd6cd460e437ab5485533a0a62b0262e15ffc421/ccpp/suites)) and also in [WM v3.0.0](https://github.com/NOAA-EMC/fv3atm/tree/release/public-v3/ccpp/suites) (which was tagged ahead of the SRW App v2.0.0 release but not "officially" considered a WM release). In general, FV3_GFS_v16 has replaced FV3_GFS_v16beta, so unless there's a reason you really need FV3_GFS_v16beta, I would suggest using the FV3_GFS_v16 suite with either the v3.0.0 or develop code. Regardless, if you switch to a different branch or tag, you will need to recompile. 

Hope this helps!

Best,
Gillian

**Reply by sanatcumar** [reply]
Hi Gillian ,
 Thanks for more detailed post . I am having issues in compiling the version I would like to use. I am on the develop branch and following the procedure listed here https://github.com/ufs-community/ufs-weather-model/blob/24c51d196ed7ec97c4256f7e78a5b0466b52937e/doc/UsersGuide/source/BuildingAndRunning.rst#L236


using the the standalone ATM configuration

export CMAKE_FLAGS="-DAPP=ATM -DCCPP_SUITES=FV3_GFS_v16"

./build.sh throws the following error
Configuring UFS app in Atmosphere Only mode

FMS .............. ON
FV3 .............. ON
AQM .............. OFF
GOCART ........... ON
MOM6 ............. ON
HYCOM ............ OFF
CICE6 ............ ON
WW3 .............. ON
STOCH_PHYS ....... ON
CDEPS ............ ON
CMEPS ............ ON
NOAHMP ........... ON

Setting configuration for hera.gnu

32BIT ............ ON
CCPP_32BIT ....... OFF
AVX2 ............. ON
SIMDMULTIARCH .... OFF
DEBUG ............ OFF
INLINE_POST ...... OFF
MULTI_GASES ...... OFF
MOVING_NEST ...... OFF
OPENMP ........... ON
PARALLEL_NETCDF .. ON
JEDI_DRIVER ...... OFF
CMEPS_AOFLUX ..... OFF

CMake Error in /scratch2/NCEPDEV/land/Sanath.Kumar/ufs-weather-model/build/CMakeFiles/CMakeTmp/CMakeLists.txt:
  Imported target "MPI::MPI_C" includes non-existent path

    "/apps/oneapi/mpi/2021.5.1/intel64/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in /scratch2/NCEPDEV/land/Sanath.Kumar/ufs-weather-model/build/CMakeFiles/CMakeTmp/CMakeLists.txt:
  Imported target "MPI::MPI_C" includes non-existent path

    "/apps/oneapi/mpi/2021.5.1/intel64/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error at /apps/cmake/3.20.1/share/cmake-3.20/Modules/FindMPI.cmake:1232 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  /apps/cmake/3.20.1/share/cmake-3.20/Modules/FindMPI.cmake:1283 (_MPI_try_staged_settings)
  /apps/cmake/3.20.1/share/cmake-3.20/Modules/FindMPI.cmake:1550 (_MPI_check_lang_works)
  CMakeLists.txt:138 (find_package)


-- Configuring incomplete, errors occurred!
See also "/scratch2/NCEPDEV/land/Sanath.Kumar/ufs-weather-model/build/CMakeFiles/CMakeOutput.log".

**Comment by gspetro-NOAA** [comment]
Hi Sanat,

It looks like you have a compiler/MPI mismatch. It says `Setting configuration for hera.gnu` but is looking for Intel MPI: `Imported target "MPI::MPI_C" includes non-existent path "/apps/oneapi/mpi/2021.5.1/intel64/include"`. 
If you ran `module load ufs_hera.intel` but are working with GNU, you would want to instead do `module load ufs_hera.gnu`. If that doesn't solve the problem, I will need you to provide a bit more information on your configuration and the specific commands you ran to understand what might have gone wrong. 

Best,
Gillian
